src/components/modules/useJsSearch.jsx

```diff
@@ -0,0 +1,28 @@
+import * as JsSearch from "js-search";
+
+// Based on https://www.dolthub.com/blog/2021-11-29-gatsby-search-and-pagination/
+
+function useJsSearch(modules) {
+  // Search configuration
+  const dataToSearch = new JsSearch.Search("name");
+  // TODO: understand how JsSearch works and which parameters we want to use here
+  dataToSearch.indexStrategy = new JsSearch.AllSubstringsIndexStrategy();
+  dataToSearch.sanitizer = new JsSearch.LowerCaseSanitizer();
+
+  // Fields to search
+  dataToSearch.addIndex("name");
+  dataToSearch.addIndex(["moduleTxt", "description"]);
+  dataToSearch.addIndex(["moduleTxt", "tags"]);
+
+  dataToSearch.addDocuments(modules);
```

Comment:
This is building the search index on the fly, specifically tailored for modules. We could potentially do that build-time, although I don't know in which phase of Gatsby that would fit best (don't want to blow up the `createPages` phase again).